Assets/SEE/Layout/NodeLayouts/IncrementalTreeMapLayout.cs

```diff
@@ -0,0 +1,354 @@
+            float width,
+            float depth,
+            IncrementalTreeMapSetting settings)
+            : base(groundLevel)
+        {
+            name = "IncrementalTreeMap";
+            _width = width;
+            _depth = depth;
+            _settings = settings;
+        }
+
+        /// <summary>
+        /// The adjustable parameters for the layout.
+        /// </summary>
+        private readonly IncrementalTreeMapSetting _settings;
+
+        /// <summary>
+        /// The width of the rectangle in which to place all nodes in Unity units.
+        /// </summary>
+        private readonly float _width;
+
+        /// <summary>
+        /// The depth of the rectangle in which to place all nodes in Unity units.
+        /// </summary>
+        private readonly float _depth;
+
+        /// <summary>
+        /// The node layout we compute as a result.
+        /// </summary>
+        private readonly Dictionary<ILayoutNode, NodeTransform> _layoutResult = new();
+
+        /// <summary>
+        /// A map to find a node (fast) by its ID
+        /// </summary>
+        private readonly Dictionary<string, Node> _nodeMap = new();
+
+        /// <summary>
+        /// A map to find a ILayoutNode (fast) by its ID
+        /// </summary>
+        private readonly Dictionary<string, ILayoutNode> _iLayoutNodeMap = new();
+
+        /// <summary>
+        /// The layout of the last revision in the evolution. Can be null.
+        /// </summary>
+        private IncrementalTreeMapLayout _oldLayout;
+
+        public IIncrementalNodeLayout OldLayout
+        {
+            set
+            {
+                if (value is IncrementalTreeMapLayout layout)
+                {
+                    this._oldLayout = layout;
+                }
+                else
+                {
+                    this._oldLayout = null;
+                    Debug.LogWarning("Incremental layout of last revision is not from same type " +
+                                     "as the layout for this revision");
+                }
+            }
+        }
+
+        public override Dictionary<ILayoutNode, NodeTransform> Layout(IEnumerable<ILayoutNode> layoutNodes)
+        {
+            var layoutNodesList = layoutNodes.ToList();
+            if (!layoutNodesList.Any()) throw new ArgumentException("No nodes to be laid out.");
+            this.roots = LayoutNodes.GetRoots(layoutNodesList);
+            InitTNodes();
+            Rectangle rectangle = new Rectangle(x: -_width / 2.0f, z: -_depth / 2.0f, _width, _depth);
+            CalculateLayout(roots, rectangle);
+            return _layoutResult;
+        }
+
+        /// <summary>
+        /// Creates and a <see cref="Node"/> for each <see cref="ILayoutNode"/>
+        /// and sets the wanted <see cref="Node.Size"/>.
+        /// Fills the <see cref="_nodeMap"/> and the <see cref="_iLayoutNodeMap"/>.
+        /// </summary>
+        private void InitTNodes()
```

Comment:
Is this a typo, or does the `T` stand for anything? If it does, the word it stands for should be in the method name, I think. Also applies to overloaded methods.